Core/Src/logger.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * logger.c
+ *
+ *  Created on: Aug 9, 2022
+ *      Author: jenish
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+#include "logger.h"
+#include "nand_types.h"
+#include "nandfs.h"
+#include "nand_m79a_lld.h"
+#include "debug.h"
+#include "iris_time.h"
+#include "command_handler.h"
+
+extern SPI_HandleTypeDef hspi1;
+
+uint8_t logger_buffer[PAGE_DATA_SIZE];
```

Comment:
It would be cleaner to combine all these in to a struct and pass the pointer to that struct to your functions